merge.go

```diff
@@ -130,18 +131,18 @@ func mergeConfigDict(opts *options, to, from *Config) Error {
 func mergeConfigArr(opts *options, to, from *Config) Error {
 	switch opts.configValueHandling {
 	case cfgReplaceValue:
-		return mergeConfigReplaceArr(opts, to, from)
+		return mergeConfigReplaceArr(fieldOptsOverride(opts, "*"), to, from)
```

Comment:
go-ucfg treats indices as fields. This looks like we will treat all indices the same. I wonder if we want support to have different merging behavior for particular indices. e.g. switch to `replace` if the setting is `settings.0`, but use `default` if the setting is `settings.1`.

One can overwrite any setting from CLI in Beats using `-E`. The use-case I have in mind is: what if the user wants to overwrite a setting in `processor.1`, but keep all others as is?

Reply:
I think we could probably expand on it in the future, but if you want me to try and implement that in this MP I can.

Reply:
I can't think of an actual use-case right now. I think this can be handled later if we really need this.

Reply:
I went ahead and added it. Unit tests actually shows a good use-case.